godtools/Managers/DownloadResources/DownloadedResourceManager.swift

```diff
@@ -52,79 +52,113 @@ class DownloadedResourceManager: GTDataManager {
             }
     }
     
-    private func saveToDisk(_ resources: [DownloadedResourceJson]) {
-        for remoteResource in resources {
-            let cachedResource = findFirstOrCreateEntity(DownloadedResource.self,
-                                                         byAttribute: "remoteId",
-                                                         withValue: remoteResource.id!)
-            
-            cachedResource.code = remoteResource.abbreviation
-            cachedResource.name = remoteResource.name
-            cachedResource.copyrightDescription = remoteResource.copyrightDescription
-            cachedResource.bannerRemoteId = remoteResource.bannerId
-            cachedResource.totalViews = remoteResource.totalViews!.int32Value
-            
-            for remoteAttachment in (remoteResource.attachments!) {
-                let remoteAttachment = remoteAttachment as! AttachmentResource
-                let cachedAttachment = findFirstOrCreateEntity(Attachment.self,
-                                                               byAttribute: "remoteId",
-                                                               withValue: remoteAttachment.id!)
-                
-                cachedAttachment.sha = remoteAttachment.sha256
-                cachedAttachment.resource = cachedResource
+    func download(_ resource: DownloadedResource) {
+        do {
+            try realm.write {
+                resource.shouldDownload = true
+                TranslationZipImporter.shared.download(resource: resource)
             }
+        } catch {
             
-            if cachedResource.bannerRemoteId != nil {
-                _ = BannerManager.shared.downloadFor(cachedResource)
+        }
+    }
+    
+    func delete(_ resource: DownloadedResource) {
+        do {
+            try realm.write {
+                resource.shouldDownload = false
+                for translation in resource.translations {
+                    translation.isDownloaded = false
+                }
+                
+                TranslationFileRemover().deleteUnusedPages()
             }
+        } catch {
             
-            let remoteTranslations = remoteResource.latestTranslations!
-            for remoteTranslationGeneric in remoteTranslations {
-                let remoteTranslation = remoteTranslationGeneric as! TranslationResource
-                let languageId = remoteTranslation.language?.id ?? "-1"
-                let resourceId = remoteResource.id ?? "-1"
-                let version = remoteTranslation.version!.int16Value
+        }
+    }
+    
+    private func saveToDisk(_ resources: [DownloadedResourceJson]) {
```

Comment:
Is it possible to breakdown this functions in other functions in order to make it easy to follow? (not sure because it is writing to Realm)